sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Statistics.scala

```diff
@@ -177,13 +180,12 @@ object ColumnStat extends Logging {
       Some(ColumnStat(
         distinctCount = BigInt(map(KEY_DISTINCT_COUNT).toLong),
         // Note that flatMap(Option.apply) turns Option(null) into None.
-        min = map.get(KEY_MIN_VALUE)
-          .map(fromExternalString(_, field.name, field.dataType)).flatMap(Option.apply),
-        max = map.get(KEY_MAX_VALUE)
-          .map(fromExternalString(_, field.name, field.dataType)).flatMap(Option.apply),
+        min = map.get(KEY_MIN_VALUE).map(fromString(_, field.name, field.dataType)),
+        max = map.get(KEY_MAX_VALUE).map(fromString(_, field.name, field.dataType)),
```

Comment:
why remove `flatMap(Option.apply)`? `fromString` may return null.

Reply:
Yea, but I tend to revert the change because keep `flatMap(Option.apply)` is more robust.